nixos/modules/services/misc/navidrome.nix

```diff
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.navidrome;
+  configFile = "/etc/navidrome/settings.json";
+  dataFolder = "/var/lib/navidrome";
+in {
+  options = {
+
+    services.navidrome = {
+      enable = mkEnableOption "Navidrome music server and streamer";
+
+      settings = lib.mkOption {
+        type = with types; attrsOf (oneOf [ int str ]);
+        default = {};
+        example = literalExample ''
+          {
+            LogLevel = "INFO";
+            BaseURL = "/music";
+            ScanInterval = "10s";
+            TranscodingCacheSize = "15MB";
+            MusicFolder = "/Media/Music";
+          };
+        '';
+        description = ''
+          Configuration for Navidrome, see <link xlink:href="https://www.navidrome.org/docs/usage/configuration-options/"/> for supported values.
+      '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "navidrome";
+        description = "User account under which Navidrome runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "navidrome";
+        description = "Group account under which Navidrome runs.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.etc."navidrome/settings.json".text = builtins.toJSON cfg.settings;
+
+    services.navidrome.settings = {
+      DataFolder = dataFolder;
+      ConfigFile = configFile;
+    };
+    
+    systemd.services.navidrome = {
```

Comment:
We could potentially use the upstream systemd unit (https://github.com/deluan/navidrome/blob/master/contrib/navidrome.service) and override options as necessary by packaging it appropriately (moving to `/lib/systemd/system` directory) in the navidrome derivation.